Cargo.toml

```diff
@@ -29,41 +29,23 @@ rust-version = "1.80"
 
 [workspace.dependencies.typespec]
 default-features = false
-path = "sdk/typespec"
```

Comment:
Is this what we decided to do? I thought we were going to leave both. This really needs to be written up somewhere and not locked in our heads. We've waffled about this too much.

That said, this is maybe premature because:

1. We should test `package-workspace` and help stabilize that. Seems it'll solve most of our problems and we won't have to have this difficult solution.
2. Until GA, we'll be rev'ing everything anyway. beta.2 is going to include a massive number of changes because I'm changing `azure_core` significantly.

Reply:
If anything, maybe we should be keeping the path dependency for the foreseeable future.